### PR TITLE
Styling of FormButtons and basics for ActionButton (WIP)

### PR DESCRIPTION
**Pull request by neongrau**
Base implementation that allows the custom layouting.

New concept for our Theming approach with Theme partials managed directly within each component.

### CODE REVIEW

File: .storybook/preview.tsx [hunk 1]
```diff
@@ -17,6 +17,58 @@ const getDefaultTheme = (): ThemeOptions => ({
...
+		main: "rgb(8, 105, 179)",
+		light: "rgb(57, 135, 194)"
+	},
+	secondary: {
+		main: "rgb(185, 215, 240)",
+	},
+};
+/* example for a customized theme per client requirements */
+const getCustomTheme = (): ThemeOptions => ({
+	palette: {
```
Comment (TQjona) [inline .storybook/preview.tsx:33]: can be simplified:
```js
palette: {
  type: "light",
  ...customColors
}
```

File: .storybook/preview.tsx [hunk 2]
```diff
@@ -84,9 +146,13 @@ const ThemeSelector = (props: ThemeSelectorProps) => {
 const AdvThemeSelector = (props: ThemeSelectorProps) => {
 	const setTheme = useContext(ThemeContext)!;
 
+	const theme = getCustomTheme();
```
Comment (TQjona) [inline .storybook/preview.tsx:149]: revert, this block sets theme every time

File: src/index.ts [hunk 3]
```diff
@@ -16,6 +18,10 @@ export * from "./utils";
 
 interface ComponentsCareTheme {
 	dataGrid?: DataGridTheme;
+	uiKit: {
```
Comment (TQjona) [inline src/index.ts:21]: `uiKit` -> `uiKit?`

File: src/standalone/UIKit/FormButtons.tsx [hunk 4]
```diff
@@ -1,18 +1,53 @@
...
+		float: theme.componentsCare?.uiKit?.formButtons?.container?.float || "left",
+		margin: theme.componentsCare?.uiKit?.formButtons?.container?.margin,
+		padding: theme.componentsCare?.uiKit?.formButtons?.container?.padding,
+		border:
+			theme.componentsCare?.uiKit?.formButtons?.container?.border || undefined,
+		borderRadius:
+			theme.componentsCare?.uiKit?.formButtons?.container?.borderRadius,
+		backgroundColor: fade(
+			theme.componentsCare?.uiKit?.formButtons?.container?.backgroundColor ||
+				"#FFF",
```
Comment (TQjona) [inline src/standalone/UIKit/FormButtons.tsx:34]: dark theme support? `theme.palette.background...`

File: src/standalone/UIKit/ActionButton.tsx [hunk 5]
```diff
@@ -24,16 +41,24 @@ const StyledButton = withStyles((theme: Theme) => ({
 			backgroundColor: theme.palette.action.disabled,
 		},
 		minWidth: props.small ? "unset" : undefined,
-		padding: props.small ? 5 : undefined,
+		padding: props.small
+			? 5
```
Comment (TQjona) [inline src/standalone/UIKit/ActionButton.tsx:45]: `theme.spacing(1)` instead of fixed 5px ?

File: src/standalone/UIKit/ActionButton.tsx [hunk 6]
```diff
@@ -24,16 +41,24 @@ const StyledButton = withStyles((theme: Theme) => ({
...
 		borderRadius: 25,
 		"&.Mui-disabled": {
+			border: theme.componentsCare?.uiKit?.actionButton?.border,
 			color: theme.palette.background.paper,
+			backgroundColor:
+				theme.componentsCare?.uiKit?.actionButton?.disabled?.backgroundColor,
 		},
-		padding: props.small ? 5 : undefined,
+		padding: props.small
+			? 5
```
Comment (TQjona) [inline src/standalone/UIKit/ActionButton.tsx:57]: `theme.spacing(1)`?

**Review by TQjona — CHANGES_REQUESTED** [review]
why is there a `yarn.lock`. we aren't using yarn.

**Comment by neongrau** [comment]
all WIP, there are quite some more details in all of these files that need to be added.
Better get the basics for theming to master sooner.

e.g. ActionButton before placed the icon as content which was a lot larger than it should be.
With the change to startIcon the small mode doesn't look right anymore, but we currently don't even need it.

The setTheme part in general doesn't work as i want it. But at least i can use it somehow. Going to change that tomorrow anyhow.

**Comment by TQjona** [comment]
if WIP => 
image

**Pull request closed without merging**